### ISSUE TITLE
Cache declined token requests

### ISSUE THREAD

**Issue by BinaryMuse** [atom/github#910]
System calls to access the stored GH token through keytar often pops up a prompt asking the user for permission to access it. If they decline this access request, then each time we look up the password (any time the `GithubLoginModel` is queried) we'll generate another request.

We should detect if we are unable to get the password and cache this fact and instead of prompting automatically make some UI that the user can click to re-query keytar.

**Comment by maxbrunsfeld** [comment]
The problem I was seeing was actually related to *accepted* token requests. Can you cache those too? I know I probably should have hit `Always Allow`, but in other apps, I'm used to just clicking `Allow`, and then not being prompted for the rest of my session.

**Comment by BinaryMuse** [comment]
Yeah, though it'd be nice to have a way to clear that cache in case the data changes outside Atom I suppose
